pkg/reconciler/apis/apibinding/apibinding_reconcile.go

```diff
@@ -47,159 +42,43 @@ const (
-			apiResourceSchema, err := r.getAPIResourceSchema(apiExportClusterName, schemaName)
-			if err != nil && apierrors.IsNotFound(err) {
-				conditions.MarkFalse(
-					apiBinding,
-					apisv1alpha1.APIExportValid,
-					apisv1alpha1.APIResourceSchemaNotFoundReason,
-					conditionsv1alpha1.ConditionSeverityError,
-					"Associated APIResourceSchema %s|%s not found",
-					apiExportClusterName,
-					schemaName,
-				)
-
-				// No way to proceed from here - don't retry
-				return reconcileStatusStop, nil
-			} else if err != nil {
-				return reconcileStatusStop, err
-			}
-
-			exportedSchemas = append(exportedSchemas, apiResourceSchema)
-		}
-
-		if apiExportLatestResourceSchemasChanged(apiBinding, exportedSchemas) {
-			klog.V(2).Infof("APIBinding %s|%s needs rebinding because the APIExport's latestResourceSchemas has changed", apiBinding.ClusterName, apiBinding.Name)
-
-			apiBinding.Status.Phase = apisv1alpha1.APIBindingPhaseBinding
-
-			// Commit the phase change, then wait for next update event to process binding changes
-			return reconcileStatusStop, nil
-		}
-	}
-
-	return reconcileStatusContinue, nil
-}
-
-func getAPIExportClusterName(apiBinding *apisv1alpha1.APIBinding) (logicalcluster.LogicalCluster, error) {
-	parent, hasParent := logicalcluster.From(apiBinding).Parent()
-	if !hasParent {
-		return logicalcluster.LogicalCluster{}, fmt.Errorf("a APIBinding in %s cannot reference a workspace", logicalcluster.From(apiBinding))
-	}
-	if apiBinding.Spec.Reference.Workspace == nil {
-		return logicalcluster.LogicalCluster{}, fmt.Errorf("APIBinding does not specify an APIExport")
-	}
-	return parent.Join(apiBinding.Spec.Reference.Workspace.WorkspaceName), nil
-}
-
-func referencedAPIExportChanged(apiBinding *apisv1alpha1.APIBinding) bool {
-	// Can't happen because of OpenAPI, but just in case
-	if apiBinding.Spec.Reference.Workspace == nil {
-		return false
-	}
-
-	if apiBinding.Status.BoundAPIExport == nil {
-		return false
-	}
-	if apiBinding.Status.BoundAPIExport.Workspace == nil {
-		return false
+		return c.reconcileBound(ctx, apiBinding)
+	default:
+		klog.Errorf("Invalid phase %q for APIBinding %s|%s", apiBinding.Status.Phase, apiBinding.ClusterName, apiBinding.Name)
+		return nil
 	}
-
-	return *apiBinding.Spec.Reference.Workspace != *apiBinding.Status.BoundAPIExport.Workspace
 }
 
-func apiExportLatestResourceSchemasChanged(apiBinding *apisv1alpha1.APIBinding, exportedSchemas []*apisv1alpha1.APIResourceSchema) bool {
-	exportedSchemaUIDs := sets.NewString()
-	for _, exportedSchema := range exportedSchemas {
-		exportedSchemaUIDs.Insert(string(exportedSchema.UID))
-	}
-
-	boundSchemaUIDs := sets.NewString()
-	for _, boundResource := range apiBinding.Status.BoundResources {
-		boundSchemaUIDs.Insert(boundResource.Schema.UID)
-	}
-
-	return !exportedSchemaUIDs.Equal(boundSchemaUIDs)
-}
+func (c *controller) reconcileNew(ctx context.Context, apiBinding *apisv1alpha1.APIBinding) error {
```

Comment:
(no action necessary) am curious why you moved to this new style of method on `controller`.